pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,477 @@
+
+		provisionedNodes, err := findNodesForNodePool(ctx, client, testNodePoolName)
+		expect.NoError(err, "Failed to find provisioned nodes")
+		Expect(len(provisionedNodes)).To(BeNumerically(">", 0), "No nodes were provisioned")
+
+		for _, node := range provisionedNodes {
+			validateAWSIntegration(node)
+		}
+	})
+
+	ginkgo.It("schedules workloads on NodePool nodes", func(ctx context.Context) {
+		nodePoolNodes, err := findReadyNodePoolNodes(ctx, client)
+		expect.NoError(err, "Failed to find ready NodePool nodes")
+		Expect(len(nodePoolNodes)).To(BeNumerically(">", 0), "No ready NodePool nodes available")
+
+		targetNode := &nodePoolNodes[0]
+		pod := createTestPod(h.CurrentProject(), targetNode.Name)
+
+		err = client.Create(ctx, pod)
+		if isSTSError(err) {
+			failWithSTSError(err, "workload creation", "ec2:RunInstances")
+		}
+		expect.NoError(err, "Failed to create test pod")
+		defer client.Delete(ctx, pod)
+
+		err = wait.For(conditions.New(client).PodPhaseMatch(pod, corev1.PodSucceeded),
+			wait.WithTimeout(3*time.Minute), wait.WithInterval(5*time.Second))
+		if isSTSError(err) {
+			failWithSTSError(err, "workload scheduling", "ec2:RunInstances")
+		}
+		expect.NoError(err, "Workload failed to complete")
+	})
+
+	ginkgo.It("handles invalid configurations gracefully", func(ctx context.Context) {
+		pod := createPodWithSelector(h.CurrentProject(), map[string]string{
+			nodePoolLabel: "non-existent-nodepool",
+		})
+
+		err := client.Create(ctx, pod)
+		expect.NoError(err, "Should create pod with invalid selector")
+		defer client.Delete(ctx, pod)
+
+		time.Sleep(30 * time.Second)
+
+		podList := &corev1.PodList{}
+		err = client.WithNamespace(pod.Namespace).List(ctx, podList)
+		expect.NoError(err, "Failed to list pods")
+
+		var foundPod *corev1.Pod
+		for _, p := range podList.Items {
+			if p.Name == pod.Name {
+				foundPod = &p
+				break
+			}
+		}
+
+		Expect(foundPod).ToNot(BeNil(), "Pod should exist")
+		Expect(foundPod.Status.Phase).To(Equal(corev1.PodPending), "Pod should remain pending")
+	})
+
+	ginkgo.It("handles resource-intensive workloads", func(ctx context.Context) {
+		nodePoolNodes, err := findReadyNodePoolNodes(ctx, client)
+		expect.NoError(err, "Failed to find ready NodePool nodes")
+		Expect(len(nodePoolNodes)).To(BeNumerically(">", 0), "No ready NodePool nodes available")
+
+		targetNode := &nodePoolNodes[0]
+		pod := createResourceIntensivePod(h.CurrentProject(), targetNode.Name)
+
+		err = client.Create(ctx, pod)
+		expect.NoError(err, "Failed to create resource-intensive pod")
+		defer client.Delete(ctx, pod)
+
+		err = wait.For(conditions.New(client).PodPhaseMatch(pod, corev1.PodSucceeded),
+			wait.WithTimeout(5*time.Minute), wait.WithInterval(5*time.Second))
+		expect.NoError(err, "Resource-intensive workload failed")
+	})
+
+	ginkgo.It("rejects operations on non-existent NodePools", func(ctx context.Context) {
+		if clusterNamespace == "" || !canAccessNodePoolAPI(h, clusterNamespace) {
```

Comment:
Once here, `clusterNamespace` should never by an empty string, should it?  And, can the `canAccessNodePoolAPI` test be done before any tests are run?